app/Http/Controllers/Auth/LoginController.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Http\Controllers\Auth;
+
+use Inertia\Inertia;
+use Inertia\Response;
+use Illuminate\Http\Request;
+use Illuminate\Support\Collection;
+use App\Http\Controllers\Controller;
+use Illuminate\Support\Facades\Route;
+
+class LoginController extends Controller
+{
+    /**
+     * Display the login view.
+     *
+     * @param Request $request
+     * @return \Inertia\Response
+     */
+    public function __invoke(Request $request): Response
+    {
+        /** @var Collection $providers */
+        $providers = config('auth.login_providers');
+        $providersName = [];
+        foreach ($providers as $provider) {
+            if ($name = config("services.$provider.name")) {
+                $providersName[$provider] = $name;
+            }
+        }
+
+        return Inertia::render('Auth/Login', [
+            'canResetPassword' => Route::has('password.request'),
+            'status' => session('status'),
+            'providers' => $providers,
```

Comment:
What happens if this is not set in the env file? It'll be an empty array?

Reply:
Exactly, I've fixed it.